tests/expected/derive-arbitrary/box/test.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that Kani can automatically derive `Arbitrary` on a struct with a
+//! member of type `Box<T>`
+
+#[derive(kani::Arbitrary)]
+struct Foo<T> {
+    boxed: Box<T>,
+}
+
+#[kani::proof]
+fn main() {
+    let foo: Foo<i32> = kani::any();
+    kani::cover!(*foo.boxed >= i32::MIN && *foo.boxed <= i32::MAX);
```

Comment:
It might be better to cover specific corner values, e.g. `kani::cover!(*foo.boxed == i32::MIN)`. Same for 0 and `i32::MAX`.

Reply:
Done: split this cover statement into the three cases you mentioned.